src/blog/2023/005-form-js-1.x.md

```diff
@@ -0,0 +1,111 @@
+---
+
+title: Empowering complex forms with form-js@1.x
+description: "Introducing recent form-js@1.x versions: These releases offer improved layouting, dynamic data binding, and FEEL expression properties to simplify complex form development."
+preview_image: http://bpmn.io/assets/attachments/blog/2023/005-group-path.png
+layout: blogpost.hbs
+slug: 2023-form-js-1-complex-form-development
+author:
+- Niklas Kiefer <https://github.com/pinussilvestrus>
+published: 2023-10-10 18:00
+
+releases:
+  - 'form-js@1.0.0'
+  - 'form-js@1.4.0'
+
+---
+
+<p class="introduction">
+  We are excited to inform about the latest [form-js@1.x releases](https://github.com/bpmn-io/form-js). 
+  These releases significantly improve the development of complex forms by enhancing layouting, dynamic data binding, and utilizing FEEL expression properties.
+</p>
+
+<!-- continue -->
+
+We dedicate the theme to helping you build complex and reusable forms with ease. Try out the new features [on our demo](https://demo.bpmn.io/form) or using the [Camunda form playground](https://github.com/camunda/form-playground).
+
+## New layouting capabilities
+
+With the addition of group components, you can now organize form components seamlessly. You can add and arrange other form components into the group, like in the parent form.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/005-group.png">
+
+  <p class="caption">
+    Create groups to organize form components logically.
+  </p>
+</div>
+
+Resizable form fields allow you to adjust the size of form components, creating visually appealing form layouts.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/005-resize-field.gif">
+
+  <p class="caption">
+    Resize form fields to fit the layout.
+  </p>
+</div>
+
+Additionally, introducing the spacer component enables you to create white space between form components, making it possible to place them more freely.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/005-spacer.png">
+
+  <p class="caption">
+    Use spacers to align form components to the right.
+  </p>
+</div>
+
+## New data binding capabilities
+
+Data binding receives a significant upgrade, expanding the possibilities of providing nested form keys.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/005-nested-key.png">
+
+  <p class="caption">
+    Use nested form keys to bind structured data.
+  </p>
+</div>
+
+Binding groups to a data path enables the provision of additional context and enhances the efficiency of organizing form data.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/005-group-path.png">
+
+  <p class="caption">
+    Specify a data path to a group to provide more context.
+  </p>
+</div>
+
+Without manual work, dynamic lists enable the effortless generation of repetitive sets of form components, such as data tables or ordered lists.
```

Comment:
This one also deserves a little rewrite.

Also, I believe we should mention that we generate magic variables in the feel contexts now that we can use to access data within the same repetition, I can write something for that.

Reply:
Or if we don't want to go too deep in the technical, we can link a future documentation page "Learn to work with dynamic lists". Although I'm not sure how okay we are with linking the Camunda docs from the bpmn-io forum.

Reply:
I believe it's okay to cross-reference Camunda documentation when it provides helpful additions 👍

Reply:
What we could also link is example diagrams that show certain behaviors in use. I also consider to replace the stock form-js viewer in the demo with the playground; maybe we can even link examples there?